lib/OpenQA/BuildResults.pm

```diff
@@ -123,29 +123,34 @@ sub compute_build_results {
     }
 
     # split the statement - give in to perltidy
-    my $search_opts = {
-        select => ['VERSION', 'BUILD', {min => 't_created', -as => 'first_hit'}],
-        as     => [qw(VERSION BUILD first_hit)],
-        # order-by is required here despite it is not relevant when iterating results
-        # because the ordering must be applied before the row limit is applied
-        order_by => {-desc => 'first_hit'},
-        group_by => [qw(VERSION BUILD)]};
-    $search_opts->{rows} = $limit if defined($limit);
-    my $search_filter = {group_id => {in => $group_ids}};
+    my %search_opts = (
+        select   => [qw(VERSION BUILD)],
+        group_by => [qw(VERSION BUILD)],
+        rows     => (defined($limit) && $limit > 400) ? $limit : 400
```

Comment:
Why? Wouldn't this prevent having a larger limit than 400? But I'm not sure whether this limit makes sense at all. Suppose we actually have more than 400 builds. Then the newest and hence most interesting will likely be last and omitted.

What is the 2nd issue?